src/modules/uxrce_dds_client/dds_topics.h.em

```diff
@@ -157,7 +157,7 @@ bool RcvTopicsPubs::init(uxrSession *session, uxrStreamId reliable_out_stream_id
 @[    for idx, sub in enumerate(subscriptions)]@
 	{
 			uint16_t queue_depth = uORB::DefaultQueueSize<@(sub['simple_base_type'])_s>::value * 2; // use a bit larger queue size than internal
-			ret = create_data_reader(session, reliable_out_stream_id, best_effort_in_stream_id, participant_id, @(idx), client_namespace, "@(sub['topic_name'])", "@(sub['dds_type'])", queue_depth);
+			ret = ret & create_data_reader(session, reliable_out_stream_id, best_effort_in_stream_id, participant_id, @(idx), client_namespace, "@(sub['topic_name'])", "@(sub['dds_type'])", queue_depth);
```

Comment:
BItwise and is not nice for this (consider e.g. 1 & 2). Or, first iteration returning 1 and second 0...

Reply:
I fixed that in 1.15.3 rebase branch like this: https://github.com/tiiuae/px4-firmware/blob/cb8a1a827a2d848dc9fa87cd20e121c62c7a2b84/src/modules/uxrce_dds_client/dds_topics.h.em#L186-L201

Reply:
I further squashed this fix into the original patch:
70aafaea23c128be371aba032db814259f244954